app/api/v1/resources_on_behalf.rb

```diff
@@ -0,0 +1,48 @@
+require 'services/publish_on_behalf_interactor'
+
+module API
+  module V1
+    # Default options for all API endpoints and versions
+    module ResourcesOnBehalf
+      extend ActiveSupport::Concern
+
+      included do
+        desc 'Takes a resource and an organization id, signs the resource '\
+             'on behalf of an organization, and publishes a new envelope with '\
+             'that signed resource',
+             http_codes: [
+               { code: 201, message: 'Envelope created' },
+               { code: 200, message: 'Envelope updated' }
+             ]
+
+        params do
+          use :update_if_exists
+          use :skip_validation
+        end
+
+        post 'resources/organizations/:organization_id/documents' do
+          authenticate!
+
+          params[:envelope_community] = select_community
+
+          interactor = PublishOnBehalfInteractor.call(
+            envelope_community: params[:envelope_community],
+            organization_id: params[:organization_id],
+            current_user: current_user,
+            raw_resource: request.body.read
+          )
+
+          error!(*interactor.error) if interactor.error
+
+          if interactor.builder_errors
```

Comment:
`builder` is just an internal implementation detail of the interactor, shouldn't bubble up to the action. Here we just want an `envelope`. Also we just want errors from the publish, regardles if they came from the builder or other verification. 
So this can be: 
```
interactor = PublishOnBehalfInteractor.call(...)
if interactor.success?
  present interactor.envelope, with: API::Entities::Envelope
  update_if_exists? ? status(:ok) : status(:created)
else
  error!(*interactor.error)
end
```
where interactor error => [msg, status]